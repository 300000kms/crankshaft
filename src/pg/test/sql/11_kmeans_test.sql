\pset format unaligned
\set ECHO all

-- spatial kmeans
SELECT
    count(DISTINCT cluster_no) as clusters
FROM
    cdb_crankshaft.cdb_kmeans('select * from ppoints', 2);

-- weighted mean
SELECT
    count(*) clusters
FROM (
    SELECT
        cdb_crankshaft.CDB_WeightedMean(the_geom, value::NUMERIC),
        code
    FROM ppoints
    GROUP BY code
) p;

-- nonspatial kmeans
SELECT
    cluster_label,
    cluster_center,
    silhouettes,
    inertia,
    rowid
FROM cdb_crankshaft.CDB_KMeansNonspatial(
    'select unnest(Array[1, 1, 10, 10]) as col1, unnest(100, 100, 2, 2) as col2 from ppoints',
    Array['col1', 'col2']::text[],
    2);
