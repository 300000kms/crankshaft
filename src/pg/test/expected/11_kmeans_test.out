\pset format unaligned
\set ECHO all
SELECT
    count(DISTINCT cluster_no) as clusters
FROM
    cdb_crankshaft.cdb_kmeans('select * from ppoints', 2);
clusters
2
(1 row)
SELECT
    count(*) clusters
FROM (
    SELECT
        cdb_crankshaft.CDB_WeightedMean(the_geom, value::NUMERIC),
        code
    FROM ppoints
    GROUP BY code
) p;
clusters
52
(1 row)
SELECT
    cluster_label,
    cluster_center,
    silhouettes,
    inertia,
    rowid
FROM cdb_crankshaft.CDB_KMeansNonspatial(
    'select unnest(Array[1, 1, 10, 10]) as col1, unnest(100, 100, 2, 2) as col2 from ppoints',
    Array['col1', 'col2']::text[],
    2);
cluster_label|cluster_center|silhouettes|inertia|rowid
0|'{"col1":1,"col2":100}'||0|1
0|'{"col1":1,"col2":100}'||0|2
1|'{"col1":10,"col2":2}'||0|3
1|'{"col1":10,"col2":2}'||0|4
(4 rows)
